conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/AsyncTargetGeneratedService.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.conjure.java.undertow.lib.BinaryResponseBody;
+import com.palantir.tokens.auth.AuthHeader;
+
+// TODO(ckozak): Delete and replace with generated code when the generator is ready
+public interface AsyncTargetGeneratedService {
+    ListenableFuture<String> string(AuthHeader authHeader);
+
+    ListenableFuture<Integer> integer(AuthHeader authHeader);
+
+    ListenableFuture<BinaryResponseBody> binary(AuthHeader authHeader);
```

Comment:
so on which thread does `BinaryResponseBody#write` end up being called?

I guess this is back on an io thread?

Reply:
`BinaryResponseBody#write` is always executed on the worker pool, blocking I/O is not allowed on the I/O threads, they're for non-blocking I/O only.

Reply:
A slow client will cause `BinaryResponseBody#write` to take a very long time, so it must run on the large pool.

Reply:
Ok - yeah - that makes sense. I now see where we dispatch back to the worker pool.

Reply:
just in https://github.com/palantir/conjure-java/pull/361/files#diff-c3b5cc874401c390b8d96224a8786de5R70 ?

Reply:
Yep, exactly